Description: Upstream changes introduced in version 0.2.32-0ubuntu3
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 unity (0.2.32-0ubuntu3) maverick; urgency=low
 .
   * Backport Gord's changes to fix scrolling and click issues in the
     launcher and Neil's changes for the recent artwork changes
   * debian/source:
     - use format 3
     - list binaries that changed in the backport
 .
 The person named in the Author field signed this changelog entry.
Author: Sebastien Bacher <seb128@ubuntu.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- unity-0.2.32.orig/unity-private/places/places-place-home-renderer.vala
+++ unity-0.2.32/unity-private/places/places-place-home-renderer.vala
@@ -94,7 +94,7 @@ namespace Unity.Places
         activate_place (APPS_PLACE, 5);
       });
 
-      icon = new HomeButton (_("Files & Folders"), filename_for_icon ("work"), "");
+      icon = new HomeButton (_("Files & Folders"), filename_for_icon ("filesandfolders"), "");
       icon_view.add_actor (icon);
       icon.show ();
       icon.clicked.connect (() => {
@@ -168,28 +168,8 @@ namespace Unity.Places
 
     private string filename_for_icon (string icon)
     {
-      if (theme == null)
-        {
-          theme = new Gtk.IconTheme ();
-          theme.set_custom_theme ("unity-icon-theme");
-        }
-
-      string icon_file = "";
-      if (theme.has_icon (icon))
-        {
-          var info = theme.lookup_icon (icon, 128, 0);
-          if (info != null)
-            {
-              var filename = info.get_filename ();
-              if (FileUtils.test(filename, FileTest.IS_REGULAR))
-                {
-                  icon_file = filename;
-                }
-            }
-
-        }
-
-      return icon_file;
+      /* FIXME: We'll autoget this from the theme before release */
+      return "/usr/share/icons/unity-icon-theme/apps/128/" + icon + ".svg";
     }
   }
 
--- unity-0.2.32.orig/unity-private/places/places-volume-child-controller.vala
+++ unity-0.2.32/unity-private/places/places-volume-child-controller.vala
@@ -24,7 +24,7 @@ namespace Unity.Places
 {
   public class VolumeChildController : ScrollerChildController
   {
-    public static const string ICON = "/usr/share/unity/trash.png";
+    public static const string ICON = "/usr/share/unity/devices.png";
 
     public Volume volume { get; construct; }
 
@@ -37,22 +37,7 @@ namespace Unity.Places
     construct
     {
       name = volume.get_name ();
-      
-      var icon = volume.get_icon ();
-      var icon_name = "";
-      if (icon is ThemedIcon)
-        {
-          icon_name = (icon as ThemedIcon).get_names ()[0];
-        }
-      else if (icon is FileIcon)
-        {
-          icon_name = (icon as FileIcon).get_file ().get_path ();
-        }
-      else
-        {
-          icon_name = ICON;
-        }
-      load_icon_from_icon_name (icon_name);
+      load_icon_from_icon_name (ICON);
 
       child.group_type = ScrollerChild.GroupType.DEVICE;
       child.drag_removed.connect (eject_volume);
